File: phoenix/lib/cards_project/cards/deck_tag.ex
defmodule CardsProject.Cards.DeckTag do
  use Ecto.Schema
  import Ecto.Changeset

  schema "deck_tags" do
    field :name, :string
    field :color, :string

    timestamps()
  end

  @doc false
  def changeset(record, attrs) do
    record
    |> cast(attrs, [:name, :color])
    |> validate_required([:name])
  end

  # ── Business operations ────────────────────────────────────────────

  def rename(_record, _new_name) do
    # TODO: implement DeckTag.rename
    :ok
  end

  def merge_into(_record, _target_tag_id) do
    # TODO: implement DeckTag.merge_into
    :ok
  end
end

File: dotnet/CardsProject/Domain/Cards/DeckTag.cs
namespace CardsProject.Domain.Cards;

public class DeckTag
{
    public int Id { get; set; }

    public string Name { get; set; } = "";
    public string? Slug { get; set; }
    public string? Color { get; set; }

    public ICollection<DeckTagAssignment> DeckAssignments { get; set; } = new List<DeckTagAssignment>();

    // Business operations

    public void Rename(string newName)
    {
        // TODO: implement rename
    }

    public void MergeInto(int targetTagId)
    {
        // TODO: implement merge_into
    }
}
